tools/autograd/derivatives.yaml

```diff
@@ -1,5 +1,355 @@
-# TODO: define derivatives e.g.
-# - name: addbmm(Scalar beta=1, Tensor self, Scalar alpha=1, Tensor mat1, Tensor mat2)
-#   self: grad * alpha
-#   mat1: grad.bmm(mat2.transpose(1, 2)) * beta
-#   mat2: mat1.transpose(1, 2).bmm(grad) * beta
+# Defines derivative formulas and Python signatures of methods on Variable
+
+- name: __and__
+- name: __iand__
+- name: __ilshift__
+- name: __ior__
+- name: __irshift__
+- name: __ixor__
+- name: __lshift__
+- name: __or__
+- name: __rshift__
+- name: __xor__
+
+- name: abs(Tensor self)
+  self: grad * self.sign()
+
+- name: acos(Tensor self)
+  self: grad * -((-self * self + 1).sqrt().reciprocal())
+
+- name: add(Tensor self, Scalar other, *, Scalar alpha=1)
+  self: grad
+
+- name: add(Tensor self, Tensor other, *, Scalar alpha=1)
+  aten: add(self, alpha, other)
+  self: grad
+  other: grad * alpha
+
+- name: addbmm(Tensor self, Tensor batch1, Tensor batch2, *, Scalar beta=1, Scalar alpha=1)
+  aten: addbmm(beta, self, alpha, batch1, batch2)
+  self: grad * alpha
```

Comment:
We can always use `grad` for single output things just for clarity, and `gradN` in case of multiple outputs (kind of like we automatically number backward functions). On the other hand I'm not sure if it will ever be needed - in most cases the additional outputs are things like indices, that are non-differentiable, and they will be leaves once we stop recording graphs for things that don't require grad.

Reply:
It comes up in double backwards -- I don't know if we would ever write those directly here, though.